File: app/src/main/java/vgame/ir/data/local/entity/CourseEntity.java
package vgame.ir.data.local.entity;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import java.util.List;

@Entity(tableName = "course")
public class CourseEntity {

    @PrimaryKey
    @SerializedName("id")
    private int id;

    @SerializedName("category_id")
    private int category_id;

    @SerializedName("name")
    private String name;

    @SerializedName("description")
    private String description;

    @SerializedName("url")
    private String url;

    @SerializedName("bookmark")
    private boolean bookmark;

    @SerializedName("teacher_id")
    private String teacher_id;

    @SerializedName("teacher_name")
    private String teacher_name;

    @Ignore
    @SerializedName("chapters")
    private List<ChapterEntity> chapters;

    @SerializedName("members_count")
    private Integer members_count;

    @Ignore
    @SerializedName("title")
    private String title;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Integer getMembers_count() {
        return members_count;
    }

    public void setMembers_count(Integer members_count) {
        this.members_count = members_count;
    }

    public String getTeacher_id() {
        return teacher_id;
    }

    public String getTeacher_name() {
        return teacher_name;
    }

    public void setTeacher_name(String teacher_name) {
        this.teacher_name = teacher_name;
    }

    public void setTeacher_id(String teacher_id) {
        this.teacher_id = teacher_id;
    }

    public boolean isBookmark() {
        return bookmark;
    }

    public void setBookmark(boolean bookmark) {
        this.bookmark = bookmark;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public List<ChapterEntity> getChapters() {
        return chapters;
    }

    public void setChapters(List<ChapterEntity> chapters) {
        this.chapters = chapters;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getUni_id() {
        return category_id;
    }

    public void setUni_id(int category_id) {
        this.category_id = category_id;
    }

    public int getCategory_id() {
        return category_id;
    }

    public void setCategory_id(int category_id) {
        this.category_id = category_id;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }
}

File: app/src/main/java/vgame/ir/data/local/entity/CourseEntity.kt
package vgame.ir.data.local.entity

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

import com.google.gson.annotations.SerializedName

@Entity(tableName = "course")
class CourseEntity {

    @PrimaryKey
    @SerializedName("id")
    var id: Int = 0

    @SerializedName("category_id")
    var uni_id: Int = 0

    @SerializedName("name")
    var name: String? = null

    @SerializedName("description")
    var description: String? = null

    @SerializedName("url")
    var url: String? = null

    @SerializedName("bookmark")
    var isBookmark: Boolean = false

    @SerializedName("teacher_id")
    var teacher_id: String? = null

    @SerializedName("teacher_name")
    var teacher_name: String? = null

    @Ignore
    @SerializedName("chapters")
    var chapters: List<ChapterEntity>? = null

    @SerializedName("members_count")
    var members_count: Int? = null

    @Ignore
    @SerializedName("title")
    var title: String? = null

    fun getCategory_id(): Int {
        return uni_id
    }

    fun setCategory_id(category_id: Int) {
        this.uni_id = category_id
    }
}